### PR TITLE
fix: add log to silent failling actions

### PR DESCRIPTION
**Pull request by alexandreliberato**
**Description:** 
Fixing silent failing actions by adding debug messages.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23570

**Testing:** 

**Documentation:**

### CODE REVIEW

File: receiver/statsdreceiver/receiver.go [hunk 1]
```diff
@@ -108,10 +109,16 @@ func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
 				batchMetrics := r.parser.GetMetrics()
 				for _, batch := range batchMetrics {
 					batchCtx := client.NewContext(ctx, batch.Info)
-					r.Flush(batchCtx, batch.Metrics, r.nextConsumer)
+					err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer)
+					if err != nil {
```
Comment (songy23) [inline receiver/statsdreceiver/receiver.go:113]: nit:
```suggestion
					if err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer); err != nil {
```

File: receiver/statsdreceiver/receiver.go [hunk 2]
```diff
@@ -108,10 +109,16 @@ func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
...
+						r.reporter.OnDebugf("Error flushing metrics", zap.Error(err))
+					}
 				}
 			case metric := <-transferChan:
-				_ = r.parser.Aggregate(metric.Raw, metric.Addr)
+				err := r.parser.Aggregate(metric.Raw, metric.Addr)
+				if err != nil {
```
Comment (songy23) [inline receiver/statsdreceiver/receiver.go:119]:
```suggestion
				if err := r.parser.Aggregate(metric.Raw, metric.Addr); err != nil {
```

File: receiver/statsdreceiver/receiver.go [hunk 3]
```diff
@@ -108,10 +109,16 @@ func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
 				batchMetrics := r.parser.GetMetrics()
 				for _, batch := range batchMetrics {
 					batchCtx := client.NewContext(ctx, batch.Info)
-					r.Flush(batchCtx, batch.Metrics, r.nextConsumer)
+					err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer)
```
Comment (songy23) [inline receiver/statsdreceiver/receiver.go:112]: This line needs to be removed

File: receiver/statsdreceiver/receiver.go [hunk 4]
```diff
@@ -108,10 +109,16 @@ func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
...
+					if err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer); err != nil {
+						r.reporter.OnDebugf("Error flushing metrics", zap.Error(err))
+					}
 				}
 			case metric := <-transferChan:
-				_ = r.parser.Aggregate(metric.Raw, metric.Addr)
+				err := r.parser.Aggregate(metric.Raw, metric.Addr)
```
Comment (songy23) [inline receiver/statsdreceiver/receiver.go:118]: Same

**Review by songy23 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**